### ISSUE TITLE
Engine possibly too powerful

### ISSUE THREAD

**Issue by gilbertohasnofb** [c172p-team/c172p-detailed#909]
Stuart wrote in a private e-mail about a potential bug with the aircraft's engine:

> [...] I've spotted a possible bug in the aircraft.  It doesn't seem to be a regression, but might be worth looking at.
> 
> Using the Fair Weather scenario (which has 20C temperatures), taking off at full throttle and puting the autopilot into a 500fpm climb, I'm seeing 145kts airspeed and the prop at 3200rpm.  This is with the 160hp variant.  Those figures seem very high, particularly in a climb.

I can confirm I can reproduce these, and actually this bug does not involve the autopilot at all: simply take off with fair weather at full throttle and manually hold a climb rate of around 500 fpm and the speed will increase _very_ fast.

**Comment by gilbertohasnofb** [comment]
@dany93 could you help me figuring out what is going on?

**Comment by dg-505** [comment]
What's the weight of the aircraft when you see these values? I think that would be important to know. At the same climb rate and with the same engine, having no pax and low fuel will of course result in a faster airspeed than climbing at MTOM... :wink:
Also: how do different weights effect the airspeed at a certain climb rate?

**Comment by gilbertohasnofb** [comment]
- full tank + pilot only = ~140 kts @ 500 fpm
- full tank + pilot + 3 pax (max weight) = ~130 kts @ 500 fpm

I agree with Stuart, these figures do look very high

**Comment by Juanvvc** [comment]
If a remember correctly, in real life, hot day, sea level, full tank, 3 pax... the C172 can't climb @ 500 fpm with a speed of 140 knots.

If have checked the procedures we used in our aircraft to take-off. Full throttle, flaps 0, rotation 60KIAS, first climbing 70KIAS. After 500AGL, continue climbing at 80 KIAS. This usually involved setting the throttle to ~80% above 500AGL.  In our airport (LERS, altitude of 250ft), fair weather, we must be above 3000ft in less than 4 minutes due to the mountains near the airport, so a real C172 climbs @500-700fpm using this procedure.

**Comment by dany93** [comment]
**POH** C172p 1982, 160 hp: (p. 92 - 93 - 94 /357 PDF)
2400 lbs
POH Rate of climb is about **600 - 700 fpm** at **75 - 76 KIAS**, between SL and 2000 ft.

**My tests**: (2400 lbs,, flaps 0, between SL and about 3000 ft))

1 - Full throttle
124 KIAS, 750 fpm but at **3055 RM** Engine. Over-regime.
(3200 RPM from Stuart still worst)
76 KIAS : 620 - 850 fpm at 2800 - 2900 RPM. Still over-regime, but less than 3055.

2 - Throttle for 2700 RPM
Throttle 0.89 - 0.90,
77 KIAS, 430 - 700 fpm (fluctuating)
_Not so far from POH specs._

3 - Cruise
2000 ft, 2500 RPM : 112 KIAS (115 KTS read on the instrument)
_In accordance with POH p. 94/357 PDF: 114 KTAS._

**Comment by gilbertohasnofb** [comment]
@dany93 does the PR sate at which RPM are those values?

**Comment by stuartbuchanan** [comment]
Hi Guys,

I'd expect that the POH rate of climb is done at full throttle.

Certainly IRL my climbs are all done at full throttle, and I'd expect that Cessna would list the best possible climb rate in their POH, which would be at full throttle.

I haven't run a test yet, but it might be interesting to run a test at full throttle and straight-and-level flight.  In my experience, you shouldn't be able to red-line the engine, nor reach Vne in straight-and-level flight.

Perhaps the prop is too fine?

-Stuart

**Comment by dany93** [comment]
@gilbertohasnofb wrote:

> @dany93 does the PR sate at which RPM are those values?

No. For the rate of climb, just "Full Throttle".

But @Juanvvc wrote:

> After 500AGL, continue climbing at 80 KIAS. This usually involved setting the throttle to ~80% above 500AGL

@stuartbuchanan wrote:
> In my experience, you shouldn't be able to red-line the engine, [...] in straight-and-level flight.

Your opinion @Juanvvc ?

And the (fixed) propeller pitch is a choice, compromise between high climb rate and high cruise velocity. Not necessarily the same for every aircraft.

Notice that the cruise values are in accordance with those from the POH.

Also, I think that these values are those from David Megginson. I don't know where he had them from, but I guess he already had a (very) good knowledge of this and of the C172P when he did it.
I can try to recalculate a propeller, but I'm not sure it will be better (and for which compromise?).
My propeller calculations are based on the [Report Nr 640 NACA curves](http://naca.central.cranfield.ac.uk/reports/1938/naca-report-640.pdf).

**Comment by stuartbuchanan** [comment]
Hi Guys,

I ran a test in straight-and-level flight at full throttle and got to VNe in <5 minutes from take-off.  I didn't check the prop RPM.  I'm sure that's not expected behaviour (in fact, I expect that airworthiness standards for GA aircraft wouldn't allow it)

Thinking about it, I wonder if the issue is with the form drag on the airframe itself rather than the engine/prop?

Given that form drag increases with the square of the speed, it may the just increasing the form drag will affect the maximum speed with minimal impact to the other speeds.

-Stuart

**Comment by HHS81** [comment]
@stuartbuchanan  
@ all 
Are you aware that the prop rpm increase much abverlangt 2400 rpm? That's what I've could see. The rpm of the prop increase a lot and goes far Beyond the red line. When I keep the rpm in a realistic Regime  (2400-2500rpm) climb rate and air speed keeps realistic. So there seems to be a issue with the engine and/ or prop.

**Comment by dany93** [comment]
I've tried an other propeller (NACA curves, 21 deg pitch).
It's much better but it can still reach over regime at full throttle if I climb or fly at too high velocity (more than 110 KIAS).

Otherwise, I have (Gross Weight 2400 lbs)
- Straight-and-level flight at 2000 ft, 2500 RPM: 115 KIAS (POH 114 KTAS),
- Climb at full throttle 75 KIAS: from 316 ft to 2626 ft in 3mn, which give 770 ft/mn average (POH between 700 ... ~ 600 fpm). Engine at full throttle was at about 2500 RPM.

Cruise velocity is good but the climb rate is a bit too high.

**Comment by dany93** [comment]
I've also found a serious issue with the engine power.
Looking at `fdm/jsbsim/propulsion/engine[0] or [1]/power-hp =`
- with the 160 hp, I see **~180 hp** at 2700 RPM,
- with the 180 hp, I see **~200 hp** at 2700 RPM. 
Although the engine tables are good for the maxhp = 160 or 180 at 2700 RPM.

It is not a general JSBSim issue, I do not have this with the DR400 (120 and 180 hp) and CAP10B.
I've never seen this before, only with our C172P at this day.

**Comment by wlbragg** [comment]
The /volumetric-efficiency is adjusted/created in System/engine.xml using a filter.
~~~xml
      </filter>
          <type>gain</type>
          <input>
             <property>/engines/active-engine/volumetric-efficiency-factor</property>
          </input>
          <output>
            <property>/fdm/jsbsim/propulsion/engine[0]/volumetric-efficiency</property>
          </output>
      </filter>
~~~

**Comment by wlbragg** [comment]
The volumetric-efficiency-factor is set in j3cub.nas a 1.
~~~xml
 setprop("/engines/active-engine/volumetric-efficiency-factor", 1.0);
~~~

**Comment by gilbertohasnofb** [comment]
@wlbragg that filter uses the property `/engines/active-engine/volumetric-efficiency-factor` which is created by the carburettor icing function, and it only _diminishes_ the efficiency. That is, `/engines/active-engine/volumetric-efficiency-factor` is always strictly less than or equal to 1.0, so I don't think it could possibly be collaborating to this issue. Just take off in non icing weather and watch that when `/engines/active-engine/carb_ice` equals to 0.0, then `/engines/active-engine/volumetric-efficiency-factor` equals always to 1.0 and so the filter is basically doing nothing.

**Comment by wlbragg** [comment]
>That is, /engines/active-engine/volumetric-efficiency-factor is always less than or equal to 1.0,

But I think that volumetric-efficiency is supposed to start at .85 and go down from there? I noticed it isn't defined in the engine definition like Erik did (aeromatic) on the Continental's in the Cub. So there may be a default that is less than one and the carb icing regime is increasing it to 1?

**Comment by wlbragg** [comment]
I was in the process of creating a branch with the volumetric-efficiency-factor set at .85 VS 1 and let them test it and see if that helps?

**Comment by wlbragg** [comment]
So this
 ~~~vml
var vol_eff_factor = 1.0 - 2.218 * carb_ice;
~~~
would need to read
~~~xml
 var vol_eff_factor = .85 - 2.218 * carb_ice;
~~~

**Comment by wlbragg** [comment]
and farther down in engine.nas
~~~xml
setprop("/engines/active-engine/volumetric-efficiency-factor", 1.0);
~~~
should read
~~~xml
setprop("/engines/active-engine/volumetric-efficiency-factor", .85);
~~~

**Comment by wlbragg** [comment]
On the Cub Continental engines it is definitely suppose to be .85 not 1. In the io320/io360 it is not defined, so there has to be a default. If you eliminate the icing regime completely (not just turn it off but remove the setting of volumetric-efficiency) we should get that default.

**Comment by gilbertohasnofb** [comment]
> But I think that volumetric-efficiency is supposed to start at .85

OMG, it's all my fault apparently!! :scream: The point is that while `/engines/active-engine/volumetric-efficiency-factor` is a made up property used by the carb icing, I am forcing its value to `/fdm/jsbsim/propulsion/engine[0]/volumetric-efficiency` instead of multiplying it......... so my intention was that `/engines/active-engine/volumetric-efficiency-factor` being at 1.0 would result in `/fdm/jsbsim/propulsion/engine[0]/volumetric-efficiency` not changing from whatever value it was, but in reality I am overwriting that value... sigh... 

I just tested removing the filter and the default value of `/fdm/jsbsim/propulsion/engine[0]/volumetric-efficiency` is indeed `0.85`. Sorry guys, a fix is coming straight away. @wlbragg thank you so much for spotting this! :1st_place_medal: 

@stuartbuchanan it seems to me that the 2017.1 release is delayed, If that is the case, would there be time for this bug fix?

**Comment by gilbertohasnofb** [comment]
@wlbragg 

> So in engine.nas this
> 
> var vol_eff_factor = 1.0 - 2.218 * carb_ice;
> 
> would need to read
> 
>  var vol_eff_factor = .85 - 2.218 * carb_ice;

Actually , the multiplier of the carb_ice should also be proportionally altered, as we intended the volumetric efficiency to be at around 33% when carb icing is at the critical engine stopping value of `0.3`. So I will rather use `var vol_eff_factor = std.max(0.0, 0.85 - 1.72 * carb_ice);`

**Comment by wlbragg** [comment]
I just verified a default of .85 if not defined!

**Comment by gilbertohasnofb** [comment]
> I just verified a default of .85 if not defined!

What do you mean?

**Comment by wlbragg** [comment]
Sorry, I missed your post!

**Comment by wlbragg** [comment]
>Oops, I missed you post where you verified this also. Oh well, now we can be sure.

I need to quit editing my posts!

**Comment by gilbertohasnofb** [comment]
@wlbragg Ok, so it is all clear. I am doing some tests now and I should push a fix very soon.

Sorry again, everyone.

**Comment by wlbragg** [comment]
No worries, I'm just glad someone noticed it. That wasn't the most obvious thing in the world.

**Comment by wlbragg** [comment]
>@wlbragg thank you so much for spotting this! :1st_place_medal:

Actually it was @dany93 that spotted it, I simply traced it after that. I would have never known that value was supposed to be .85

**Comment by gilbertohasnofb** [comment]
@wlbragg @dany93 thank to you both for spotting this. A PR has been opened

**Comment by gilbertohasnofb** [comment]
@dany93 @stuartbuchanan could you guys check that PR and see if that was all that there was to the issue? The branch name is `bug-909`. If that was all, then we can merge it.

**Comment by stuartbuchanan** [comment]
Hi Guys,

I'm not going to be able to do anything until Friday evening at the
earliest due to being on a business trip.

I think Torsten is building the release right now, so I suspect we're too
late.  However, you can try to get in touch with Torsten Dryer and ask if
there is scope for fixing this.

Best regards,

-Stuart

**Comment by gilbertohasnofb** [comment]
@stuartbuchanan Ok, thanks for the info. As soon as that PR is merged I will contact Torsten and see what he says.

**Comment by onox** [comment]
Fix in #911 backported to `release/2017.1` with `git cherry-pick -esS 3939d52d0768b395a2e422d038b51dfae4bdc83e`.

**Comment by onox** [comment]
@dany93 Good debugging! 👍 😄

**Comment by gilbertohasnofb** [comment]
> Fix in #911 backported to release/2017.1

@onox thanks :+1:

**Comment by dany93** [comment]
Sorry, I wasn't at home this afternoon.

`/fdm/jsbsim/propulsion/engine[0] and [1]/volumetric-efficiency` is back to 0.85 for both engines.

I've checked, `fdm/jsbsim/propulsion/engine[0] or [1]/power-hp` at 2700 RPM, mixture 100%, are 157 hp and 177 hp near sea level.

I had not enough time to check the  aircraft performances (velocity, climb rate) but I see no reason, everything should be back to previously.

Glad that you could fix the code and got it merged for the 2017.1 release. Thumbs up, It was short!

By the way, why is the mixture set at 95% with Autostart? It is somewhat disturbing and those who use the Autostart will almost never see it. Which prevents them from reaching full power.

**Comment by gilbertohasnofb** [comment]
@dany93 thanks for confirming that everything looks good once again.

> I had not enough time to check the aircraft performances (velocity, climb rate) but I see no reason, everything should be back to previously.

Absolutely, it is more than clear what the mistake was.

> Glad that you could fix the code and got it merged for the 2017.1 release. Thumbs up, It was short!

Well, it is still not merged into FGDATA yet and we haven't heard back from Torsten, so let's hope it will make into the 2017.1 release!

> By the way, why is the mixture set at 95% with Autostart? It is somewhat disturbing and those who use the Autostart will almost never see it. Which prevents them from reaching full power.

I believe this was done when @onox and I were improving the autostart function, the commit is https://github.com/c172p-team/c172p-detailed/pull/804/commits/3662ef5d7b101f2e1078a4f4d1deb2b57de50b8b and the discussion for this is https://github.com/c172p-team/c172p-detailed/issues/791. Basically, the reason for the decision is that with mixture at 95% and throttle at 20%, the engine will start at virtually any single altitude pressure from which a 172 can legally fly from (when the mixture is at 100%, the engine is not able to start at certain altitudes). We initially planned to use a table to relate altitudes to mixture settings but once we found out that 95% was a magic number to start the engine at virtually any condition, we took the simpler route. But that said, perhaps we should force the mixture to 100% when the altitude is below 3000 feet, what do you think?

**Comment by wlbragg** [comment]
I never realized that is what was done. I don't think it is the right thing to do unless the user is notified of that setting.
Even though more work, it makes more sense to me to use a table or something that will set the mixture only when needed to start the aircraft at the startup altitude..

**Comment by onox** [comment]
When we were experimenting with the mixture, I noticed that at higher elevations, that 100 % mixture would not start the engine at all, but at 95 % it would (which is just a tiny bit lower). It is sort of a one-size-fits-all solution, so if you want to replace it with something more accurate (a table), go ahead. You can put the table in JSBSim/XML for interpolation between values.

**Comment by gilbertohasnofb** [comment]
Great news everyone, Torsten just wrote me:

> I have added your fix to next and the release/2017.1 branch. It will make it into the 2017.1.1 release.

And I confirmed the fix is present in FGDATA :tada:
